Dockerfile

```diff
@@ -0,0 +1,22 @@
+FROM lukemathwalker/cargo-chef:latest-rust-1 AS chef
```

Comment:
Introduces new tooling, why? Is this something that we couldn't do with a simple `cargo build --release`?

Reply:
These images aren't for public consumption, at least that isn't the intention. At a minimum, the service would have to be modified to persist data rather than storing in memory to productionize it. So the image size is not that important. Just using a two stage builder and copying the binary to the final container should produce a perfectly good image.

I'm asking because each piece of tooling we introduce is something else that we have to maintain over time and anybody who has to work on or update this image will need to learn about.

I'd like to leave this thread open for a second reviewer and see what somebody else thinks. I'm happy with the rest of the PR, just waiting on more feedback for this.